src/lib/label/label-aware.ts

```diff
@@ -0,0 +1,16 @@
+export interface ILabelAware {
+  labelClickedCallback(): void;
+  labelChangedCallback(value: string | null): void;
+}
+
+/**
+ * Determines if an object is label aware.
+ * @param obj - The object to check.
+ * @returns True if the object is label aware, false otherwise.
+ */
+export const isLabelAware = (obj: any): obj is ILabelAware => {
+  return obj.labelClickedCallback &&
```

Comment:
You can probably remove the truthy evaluations here since `typeof` will handle that as well